src/Chapter3.hs

```diff
@@ -480,6 +506,37 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data MainBuilding = Library | Church deriving (Show)
+data House = OnePerson | TwoPersons | ThreePersons | FourPersons deriving (Show)
+data Castle = None | Castle String | CastleWithWalls String deriving (Show)
+
+data MagicalCity = MagicalCity
+    { cityMainBuilding :: MainBuilding
+    , cityCastle :: Castle
+    , cityHouses :: [House]
+    } deriving (Show)
+
+buildCastle :: String -> MagicalCity -> MagicalCity
+buildCastle name city = city { cityCastle = Castle name }
```

Comment:
It looks like if the city already has walls, they will be destroyed with this function :thinking:

Reply:
Fixed in 82768fe 

I thought that walls are to be demolished too if new castle is built